Add class documentation, improve comments in `MazeRecursion.java`

Hardvan opened the pull request:
<!--
Thank you for your contribution!
In order to reduce the number of notifications sent to the maintainers, please:
- create your PR as draft, cf. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests,
- make sure that all of the CI checks pass,
- mark your PR as ready for review, cf. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request#marking-a-pull-request-as-ready-for-review
-->

<!-- For completed items, change [ ] to [x] -->

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized it.
- [x] All filenames are in PascalCase.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in their comments that points to Wikipedia or other similar explanations.
- [x] All new code is formatted with `clang-format -i --style=file path/to/your/file.java`

siriak commented on src/main/java/com/thealgorithms/backtracking/MazeRecursion.java:
```diff
@@ -1,66 +1,76 @@
+     * represents open paths. It then calls recursive functions to find paths using
+     * two different movement strategies. The results are printed to the console.
+     */
     public static void mazeRecursion() {
```
To improve this class, please make methods accept maze to solve as a parameter and return the solution instead of printing it. Also add JUnit tests

Hardvan replied:
Change implemented!

Hardvan commented:
I modified the class to accept a maze as a parameter and return the solution instead of printing it. I also added JUnit tests to validate the functionality.

siriak reviewed (approved):
Looks good, thanks!
